framework/decode/dx12_dump_resources.cpp

```diff
@@ -206,16 +208,21 @@ bool Dx12DumpResources::ExecuteCommandLists(DxObjectInfo*
                     // processes, for exmaples: finding resource by GPU VA, getting the resource infomation, and
                     // write resource id, offset, size. But those duplicated processes shouldn't hurt the
                     // performance.
-                    CopyDrawCallResources(replay_object_info,
-                                          front_command_list_ids,
-                                          graphics::dx12::Dx12DumpResourcePos::kBeforeDrawCall);
+                    if (options_.dump_resources_before)
+                    {
+                        CopyDrawCallResources(replay_object_info,
+                                              front_command_list_ids,
+                                              graphics::dx12::Dx12DumpResourcePos::kBeforeDrawCall);
+                    }
 
                     ID3D12CommandList* ppCommandLists[] = { track_dump_resources_.split_command_sets[1].list };
                     replay_object->ExecuteCommandLists(1, ppCommandLists);
 
                     CopyDrawCallResources(replay_object_info,
                                           front_command_list_ids,
-                                          graphics::dx12::Dx12DumpResourcePos::kAfterDrawCall);
+                                          options_.dump_resources_before
```

Comment:
I think this would be easier to read if `kAfterDrawCall` and `kAfterDrawCall_NoBefore` were bitmasks that were or'd, `kAfterDrawCall` and `kBeforeDrawCall` unless I've misunderstood this.

Reply:
On Vulkan, when --dump-resources-before-draw is specified, the bmp/bin/png files produced have the word "before" in the filename if it is a dump before the draw call and the word "after" it is is a dump after the draw call. If  --dump-resources-before-draw is not specified, the dump occurs after the draw call and file names do not include either "before" or "after". I added the "kAfterDrawCall_NoBefore"  for the case where the option is not specified and we don't want to include "before" or "after" in the file name. A bitmask would imply that two or more of these could occur at the same time, but that it not the case. The enums describe the three different cases.